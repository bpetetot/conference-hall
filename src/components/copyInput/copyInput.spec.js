/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { mount } from 'enzyme'
import snap from 'tests/snapshot'
import CopyInput from './index.js'

const snapshot = props => snap(CopyInput)({ ...props })

describe('components/copyInput', () => {
  it('should render with all props', snapshot({ title: 't', value: 'v' }))
  it('should render without title', snapshot({ value: 'v' }))
  it('should render without props', snapshot({}))
  it('should copy the value in clipboard when button clicked', () => {
    // mock
    document.execCommand.mockReset()
    // mouting
    const wrapper = mount(<CopyInput value="to copy" />)
    // action
    wrapper.find('button').simulate('click')
    // assert
    expect(document.execCommand.mock.calls).toMatchSnapshot()
  })
})
