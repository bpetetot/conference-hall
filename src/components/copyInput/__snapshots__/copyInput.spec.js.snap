// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/copyInput should copy the value in clipboard when button clicked 1`] = `
Array [
  Array [
    "Copy",
  ],
]
`;

exports[`components/copyInput should render with all props 1`] = `
<div
  className="copy-input"
>
  <label
    className="copy-title"
    htmlFor="copy-input"
  >
    t
  </label>
  <input
    defaultValue="v"
    id="copy-input"
    onFocus={[Function]}
    placeholder="Just copy it"
    type="text"
  />
  <button
    className="cc-button cc-button-tertiary"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-clipboard"
    />
  </button>
</div>
`;

exports[`components/copyInput should render without props 1`] = `
<div
  className="copy-input"
>
  <input
    id="copy-input"
    onFocus={[Function]}
    placeholder="Just copy it"
    type="text"
  />
  <button
    className="cc-button cc-button-tertiary"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-clipboard"
    />
  </button>
</div>
`;

exports[`components/copyInput should render without title 1`] = `
<div
  className="copy-input"
>
  <input
    defaultValue="v"
    id="copy-input"
    onFocus={[Function]}
    placeholder="Just copy it"
    type="text"
  />
  <button
    className="cc-button cc-button-tertiary"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-clipboard"
    />
  </button>
</div>
`;
